src/type_checker/mod.rs

```diff
@@ -127,6 +127,7 @@ impl<B: Backend> TypeChecker<B> {
             }
             TyKind::BigInt => 1,
             TyKind::Array(typ, len) => (*len as usize) * self.size_of(typ),
+            TyKind::Generic(_) => unreachable!("generic should have been resolved"),
```

Comment:
It feels weird to have this in the type checker now. What do you think about moving this function to the MAST? (like that it really shows that this is not allowed in the MAST representation, and not in the TAST)

Reply:
All the types are represented by the `TyKind`, no matter whether it is generic or concrete.
 
Maybe we can have MAST have its own `TyKind`, but only for concrete, while the one for TAST represents both the concrete and generic type.

So the `size_of` function should be based on the MAST `TyKind`, and it should only contain concrete types.